storage-proofs/porep/src/stacked/vanilla/create_label/multi.rs

```diff
@@ -0,0 +1,681 @@
+        info!("Layer {}", layer);
+
+        // Cache reset happens in two parts.
+        // The second part (the finish) happens before each layer but the first.
+        if layers != 1 {
+            parents_cache.finish_reset()?;
+        }
+        create_layer_labels(
+            &parents_cache,
+            &replica_id.as_ref(),
+            &mut layer_labels,
+            if layer == 1 {
+                None
+            } else {
+                Some(&mut exp_labels)
+            },
+            node_count,
+            layer as u32,
+        )?;
+
+        // Cache reset happens in two parts.
+        // The first part (the start) happens after each layer but the last.
+        if layer != layers {
+            parents_cache.start_reset()?;
+        }
+
+        {
+            let layer_config =
+                StoreConfig::from_config(&config, CacheKey::label_layer(layer), Some(graph.size()));
+
+            info!("  storing labels on disk");
+            // Construct and persist the layer data.
+            let layer_store: DiskStore<<Tree::Hasher as Hasher>::Domain> =
+                DiskStore::new_from_slice_with_config(
+                    graph.size(),
+                    Tree::Arity::to_usize(),
+                    &layer_labels,
+                    layer_config.clone(),
+                )?;
+            info!(
+                "  generated layer {} store with id {}",
+                layer, layer_config.id
+            );
+
+            std::mem::swap(&mut layer_labels, &mut exp_labels);
+
+            // Track the layer specific store and StoreConfig for later retrieval.
+            labels.push(layer_store);
+            label_configs.push(layer_config);
+        }
+    }
+    assert_eq!(
+        labels.len(),
+        layers,
+        "Invalid amount of layers encoded expected"
+    );
+
+    Ok((
+        LabelsCache::<Tree> { labels },
+        Labels::<Tree> {
+            labels: label_configs,
+            _h: PhantomData,
+        },
+    ))
+}
+
+pub fn create_label(
+    cache: &mut ParentCache,
+    replica_id: &[u8],
+    exp_labels: &[u8],
+    layer_labels: &mut [u8],
+    layer_index: usize,
+    node: usize,
+    first_layer: bool,
+) -> Result<()> {
+    let mut hasher = Sha256::new();
+    let mut buffer = [0u8; 32];
+
+    hasher.update(replica_id);
+
+    buffer[0..4].copy_from_slice(&(layer_index as u32).to_be_bytes());
+    buffer[4..12].copy_from_slice(&(node as u64).to_be_bytes());
+    hasher.update(buffer);
+
+    // hash parents for all non 0 nodes
+    let hash = if node == 0 {
+        hasher.finalize()
+    } else {
+        let cache_parents = &cache.read(node as u32)?;
+
+        // prefetch previous node, which is always a parent
+        let prev = &layer_labels[(node - 1) * NODE_SIZE..node * NODE_SIZE];
+        prefetch!(prev.as_ptr() as *const i8);
+
+        if first_layer {
+            let mut parents = [0u8; BASE_DEGREE * NODE_SIZE]; // FIXME: Don't hardcode 14.
```

Comment:
FIXME

Reply:
Thanks. It was fixed, just needs comment removed. However, this is a good reminder that this is all dead code now anyway. Will remove in next push.